File: backend/app/schemas/knowledge_base.py
"""Pydantic schemas for product catalog endpoints."""

from datetime import datetime

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: int
    filename: str
    file_type: str
    file_size: int | None = None
    chunk_count: int
    status: str
    error_message: str | None = None
    uploaded_by: int | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    total: int
    page: int
    per_page: int


class UploadResponse(BaseModel):
    id: int
    filename: str
    status: str
    message: str = "Document uploaded successfully. Processing started."


class ReprocessResponse(BaseModel):
    id: int
    status: str
    message: str = "Document reprocessing started."


class ChunkPreview(BaseModel):
    chunk_index: int
    content_preview: str
    metadata: dict


class DocumentDetailResponse(DocumentResponse):
    chunks: list[ChunkPreview] = []
    chunk_total: int = 0

File: frontend/src/types/knowledgeBase.ts
export interface Document {
  id: number;
  filename: string;
  file_type: string;
  file_size: number | null;
  chunk_count: number;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  error_message: string | null;
  uploaded_by: number | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface DocumentListResponse {
  items: Document[];
  total: number;
  page: number;
  per_page: number;
}

export interface ChunkPreview {
  chunk_index: number;
  content_preview: string;
  metadata: Record<string, unknown>;
}

export interface DocumentDetail extends Document {
  chunks: ChunkPreview[];
  chunk_total: number;
}

export interface UploadResponse {
  id: number;
  filename: string;
  status: string;
  message: string;
}
